app/app.js

```diff
@@ -701,6 +684,36 @@ YUI.add('juju-gui', function(Y) {
     },
 
     /**
+      Composes the various socket paths and protocols and returns the corret
+      url that the GUI should use to communicate with the environment.
+
+      @method _generateSocketUrl
+      @return {String} The fully qualified socket url.
+    */
+    _generateSocketUrl: function() {
+      var socketProtocol = this.get('socket_protocol');
+      // Assemble a socket URL from the Location.
+      var loc = Y.getLocation();
+      socketProtocol = socketProtocol || 'wss';
+      var socketUrl = socketProtocol + '://' + loc.hostname;
+      if (loc.port) {
+        socketUrl += ':' + loc.port;
+      }
+      // If the Juju version is over 1.21 then we need to make requests to the
+      // api using the environments uuid.
+      var jujuVersion = this.get('jujuCoreVersion').split('.');
+      var suffix = '';
+      var majorVersion = parseInt(jujuVersion[0], 10);
+      var minorVersion = parseInt(jujuVersion[1], 10);
+      if (majorVersion === 1 && minorVersion > 20 || majorVersion > 1) {
```

Comment:
Yeah the UUID has actually been available for some time, much longer than the api supported it unfortunately.